### ISSUE TITLE
[WP] Allow (optionally?) a limited set of HTML markup in node titles.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5081]
This is basically a request to include the functionality of https://www.drupal.org/project/html_title into core, to allow _portions_ of the node title to be italicized/bolded/sub-superscripted etc.

> Drupal node titles are restrictive and do not support italicized text or superscript text (e.g., Book titles, trademark symbols). The HTML Title module allows a limited set of HTML markup (em, sub, sup, b, i, strong, cite, code, bdi, wbr) to be used in node titles. It filters all other markup from titles to prevent XSS vulnerabilities or irrational nesting (e.g., links within links).

**Comment by klonos** [comment]
Do people think that this is better left for contrib, or something that would annoy people when they discover that Backdrop breaks the formatting of their titles when copy-pasting them from other sources?

**Comment by klonos** [comment]
WP allows that OOTB. You can add some allowed HTML in the title when creating/editing content:

![image](https://user-images.githubusercontent.com/2423362/117332739-dbc42f80-aea0-11eb-9b71-38be63b945d3.png)

And it is rendered as expected:

![image](https://user-images.githubusercontent.com/2423362/117332813-f3031d00-aea0-11eb-9ec2-dc04e9164f75.png)

**Comment by klonos** [comment]
The resulting page (and menu items) when you attempt the same in Backdrop is this:

![image](https://user-images.githubusercontent.com/2423362/117333494-a79d3e80-aea1-11eb-8258-070dc35a058b.png)

...and path is `this-is-a-test-title-in-backdroptm`

**Comment by BWPanda** [comment]
Related: https://github.com/backdrop/backdrop-issues/issues/2050

**Comment by BWPanda** [comment]
It seems @klonos has changed his mind since [2016](https://github.com/backdrop/backdrop-issues/issues/2050#issuecomment-238422346) when he said:

> Any markup should be done in the theme layer - not stored as a text field value.

😋

**Comment by klonos** [comment]
Yeah ...I seem to have mixed feelings (or a split personality some might say) 😅 

What actually happens is that I keep forgetting that we are talking about allowing _portions_ of that text to have markup. So yes, it'd be preferred to handle markup that applies to the _entire_ title (or site name/slogan in the case of #2050), but it's really tricky to apply markup only to specific words/letters (requires 3rd-party JS libraries).
